### DISCUSSION TITLE
关于选择控制点的想法

### DISCUSSION THREAD

**Discussion by todpeng777** [huchenlei/sd-webui-openpose-editor#34] [Ideas]
终于可以使用这个插件了！！使用过程中遇到几个问题，有点不成熟的想法，大佬看看是否有用？
1，关闭按钮是否可以和主面板近一些？我在第一次使用的时候要退出的时候找了很久关闭按钮，最后才发现在一个角落里，如果把它和面板近一些，会不会更好找？
2，鼠标中键的操作，在插件面板内空白处滑动鼠标中键，会对下面主界面产生上下滑动，是不是屏蔽掉会好？
3，最主要的是选取控制点太麻烦，特别是手部，密密麻麻的，虽然有姿势控制列表，但比较长，名称又类似，不是很直观。所以我想能不能增添一个骨骼导航界面，（如图）可以直观地在骨骼导航中选取想要的那个点，并且选中的点会高亮。（当然位移操作还是在主面板中，右边的骨骼只是方便选取那些点，导航骨骼始终是完整的，隐藏的部分灰色显示，这样就知道到底哪些部分被隐藏了）
![image](https://github.com/huchenlei/sd-webui-openpose-editor/assets/125846611/28303282-8ff9-469d-ac03-e69273ec472a)

因为身体部分，手部分，脸部分的控制点密度不同，如果能把他们分为几块，是不是能更加方便选取？（如下图）把手的骨骼图配合上一张手部图片，这样就方便知道哪根手指对应哪根骨骼了
![image](https://github.com/huchenlei/sd-webui-openpose-editor/assets/125846611/d9d30365-2cda-4c74-a5af-3e7c512a5554)

4，这个使用过程，放大缩小位移是高频操作，如果把这几个操作的工具做在界面上会不会方便一点？

**Comment by huchenlei** [comment]
1. See discussion https://github.com/huchenlei/sd-webui-openpose-editor/discussions/20#discussioncomment-6145034. Current design is the same design of image preview popup modal in A1111.

2. 请截图说明是在哪个位置使用鼠标滚轮会导致主界面scroll
3. 手部关键点密集，但是可以缩放画布对细节进行调整。你可以点击关键点，左边姿势控制里对应关键点的行会红色高亮。如果选择多个关键点，那么那些对应关键点的行都会红色高亮。脸部因为关键点多而且之间都没有连接，我设计的时候其实是没打算让用户去过多自己捏脸部的。
4. 我个人使用体验下来感觉这种高频操作还是用键位绑定来操作比较合适，通过鼠标点选去改变模式我认为没有必要。如果有其他人也raise了同样的问题我会考虑加上。

**Comment by todpeng707** [comment]
没有展开选择列表的时候，鼠标放在这些标红的位置，即使是字上面，滚动鼠标中键，移动的会是下面的主界面，
![image](https://github.com/huchenlei/sd-webui-openpose-editor/assets/125846768/51429033-b59c-40a5-b14c-f5c1fbff4fd6)
展开了选择列表的时候，左侧就不会导致下层的移动，但右边红色区域鼠标中键都会移动下层主界面
![image](https://github.com/huchenlei/sd-webui-openpose-editor/assets/125846768/59f74a4d-350c-4781-bf21-b185ec9764c9)

会不会是我这边有没有什么特殊导致的？
